addons/account/tests/test_account_move_reconcile.py

```diff
@@ -4438,3 +4384,179 @@ def test_reconcile_payment_custom_rate(self):
+
+        self.env['account.payment.register'].with_context(active_model='account.move', active_ids=invoice.ids).create({
+            'payment_date': invoice.date,
+        })._create_payments()
+
+        # Check the analytic distribution on the caba move lines.
+        # I.e. lines with different analytic distributions should not be grouped / mixed together.
+        caba_move = invoice.tax_cash_basis_created_move_ids
+        expected_caba_move_line_values = [
+            {
+                'account_id': self.cash_basis_base_account.id,
+                'debit': 0.0,
+                'credit': 800.0,
+                'analytic_distribution': analytic_distribution_a_serialized,
+            },
+            {
+                'account_id': self.cash_basis_base_account.id,
+                'debit': 800.0,
+                'credit': 0.0,
+                'analytic_distribution': analytic_distribution_a_serialized,
+            },
+            {
+                'account_id': self.cash_basis_base_account.id,
+                'debit': 0.0,
+                'credit': 1600.0,
+                'analytic_distribution': analytic_distribution_b_serialized,
+            },
+            {
+                'account_id': self.cash_basis_base_account.id,
+                'debit': 1600.0,
+                'credit': 0.0,
+                'analytic_distribution': analytic_distribution_b_serialized,
+            },
+            {
+                'account_id': self.tax_account_1.id,
+                'debit': 0.0,
+                'credit': 192.0,
+                'analytic_distribution': False,
+            },
+            {
+                'account_id': self.cash_basis_transfer_account.id,
+                'debit': 192.0,
+                'credit': 0.0,
+                'analytic_distribution': False,
+            },
+            {
+                'account_id': self.cash_basis_base_account.id,
+                'debit': 0.0,
+                'credit': 96.0,
+                'analytic_distribution': analytic_distribution_a_serialized,
+            },
+            {
+                'account_id': self.cash_basis_transfer_account.id,
+                'debit': 96.0,
+                'credit': 0.0,
+                'analytic_distribution': analytic_distribution_a_serialized,
+            },
+            {
+                'account_id': self.cash_basis_base_account.id,
+                'debit': 0.0,
+                'credit': 192.0,
+                'analytic_distribution': analytic_distribution_b_serialized,
+            },
+            {
+                'account_id': self.cash_basis_transfer_account.id,
+                'debit': 192.0,
+                'credit': 0.0,
+                'analytic_distribution': analytic_distribution_b_serialized,
+            },
+        ]
+        self.assertRecordValues(caba_move.line_ids.sorted('id').sorted('sequence'),  # sorted like in the view
+                                expected_caba_move_line_values)
```

Comment:
```suggestion
        self.assertRecordValues(caba_move.line_ids.sorted('id').sorted('sequence'),  expected_caba_move_line_values)
```
Inlining is more readable in this case; and I think the comment is not very useful